### ISSUE TITLE
FDS to DCM

### ISSUE THREAD

**Issue by Alethx** [marksgraham/OCT-Converter#6]
Hi, Mark,
This is a really useful tool for imaging research. I am wondering if you could add the conversion from FDS to DCM in the future.

**Comment by Alethx** [comment]
Conversion from FDS to AVI or MP4 leads to imgae information loss, dcm would be a better choice. Anyway to do that?

**Comment by marksgraham** [comment]
Hi,
Currently the code also supports saving as numpy arrays (.npy), which will not lose you data. But I appreciate they are only useful if you want to continue analysing the data in Python. It shouldn't be hard to modify the code to save at least a basic Dicom file using Pydicom or similar - I'll take a look at this soon.

**Comment by Alethx** [comment]
That would be very nice! Thanks a lot!

**Comment by marcoduering** [comment]
Hi! DICOM export would be really helpful. Any updates in this regard? Many thanks!

**Comment by marksgraham** [comment]
Hi there,
I still haven't had time to work on this. If you had time and were keen I would be happy to take a pull request - a simple implementation would just search for .dcm extensions in the save method [here](https://github.com/marksgraham/OCT-Converter/blob/master/oct_converter/image_types/oct.py) and then initalised a dicom file from all the data available using pydicom.

**Comment by marcoduering** [comment]
Do I see correctly that you are currently not using/storing any geometry information? So e.g. no pixel scaling/dimensions? Saving the data to DICOM is one part, but writing a header with the correct info also on geometry is probably also needed.

**Comment by marksgraham** [comment]
That is right, currently I don't extract any geometry from the files. It will be possible to get this for some file types (e.g. I think we can do it for .fds) but probably not all. So having it implemented without all the header info will be useful for those who use dicom viewers to examine their data and want to be able to do that, or have pipelines that read in the imaging data from .dcm files,  but it will take much more work to get it to the point where there is much useful header information available.

**Comment by marksgraham** [comment]
Very rudimentary dicom support [here](https://github.com/marksgraham/OCT-Converter/blob/4694c67913841b8af34f9a00376acce9e713fcda/oct_converter/readers/dcm.py) and this isn't a priority for further development atm.
